projects/code_quality_agent/states/code_quality_state.py

```diff
@@ -20,4 +21,139 @@ class PRQualityAssessment(BaseModel):
 class CodeQualityState(BaseModel):
     employee: Employee = Field(description="Employee object")
     messages: List[BaseMessage] = Field(description="List of messages")
-    assessment: Optional[PRQualityAssessment] = Field(description="Code quality assessment", default=None)
\ No newline at end of file
+    assessment: Optional[PRQualityAssessment] = Field(description="Code quality assessment", default=None)
+
+
+class QualityLevel(int, Enum):
+    WORST = 0
+    BEST = 100
+
+
+class Decision(str, Enum):
+    PASS = "PASS"
+    FAIL = "FAIL"
+
+
+class Recommendation(BaseModel):
+    line_numbers: List[int] = Field(
+        description="Lines of code where the issue is identified."
+    )
+    issue: str = Field(
+        description="Description of the identified issue."
+    )
+    recommended_fix: str = Field(
+        description="Suggested fix or improvement for the identified issue."
+    )
+
+
+class BaseEvaluation(BaseModel):
+    score: QualityLevel = Field(
+        description="Score, where 0 represents WORST and 100 represents BEST."
+    )
+    threshold: int = Field(
+        description="Threshold score for passing (0-100).",
+        ge=QualityLevel.WORST,
+        le=QualityLevel.BEST
+    )
+    recommendations: Optional[List[Recommendation]] = Field(
+        description="List of recommendations for improving this aspect.",
+        default=None
+    )
+
+    @property
+    def final_decision(self) -> Decision:
+        """Calculate the final decision based on the score and threshold."""
+        return Decision.PASS if self.score >= self.threshold else Decision.FAIL
+
+
+class ReadabilityAndMaintainability(BaseEvaluation):
+    semantic_understanding: str = Field(
+        description="Assessment of semantic understanding, including intent, complexity, and logical coherence."
+    )
+    design_pattern_recognition: str = Field(
+        description="Evaluation of the use of design patterns, anti-patterns, and adherence to SOLID principles."
+    )
+    documentation_quality: str = Field(
+        description="Assessment of documentation quality, including value-added comments, missing documentation, and consistency with code."
+    )
+
+
+class Security(BaseEvaluation):
+    input_validation: str = Field(
+        description="Assessment of input validation mechanisms to prevent malicious input."
+    )
+    authentication_and_authorization: str = Field(
+        description="Evaluation of authentication and authorization practices."
+    )
+    encryption_practices: str = Field(
```

Comment:
maybe `cryptographic_weakness` because it encompasses more and encryption would be a subset of it.